README.md

```diff
@@ -1,79 +1,200 @@
+
+### Flask
+Flask is a Web Application Framework in Python. 
+We use Flask to develop the components of our web application because it helps us integrate existing libraries into our software.
+Extensibility and good documentation characterize Flask. 
+Link: [https://www.palletsprojects.com/p/flask/](https://www.palletsprojects.com/p/flask/)
+
+### TypeScript
+TypeScript is a superset of JavaScript, i.e. it extends JavaScript by features such as, 
+in particular, the optional static typing at the time of compilation. 
+This allows us to , for example, structurally type, infer types, and parameterize types. 
+Link: [https://www.typescriptlang.org](https://www.typescriptlang.org)
+
+### React
+React is a JavaScript-based web framework, i.e. it acts as a basic framework for the output of UI components on websites. 
+In React, we build components hierarchically and represent them as self-defined HTML tags. 
+React is designed for simplicity, performance, and reusability. 
+Link: [https://reactjs.org](https://reactjs.org)
+
+### Graph Visualization Frameworks
+Graph visualization frameworks let us represent structural information as networks. 
+
+#### ECharts
+ECharts is an open source, cross-platform framework for rapidly constructing data visualizations. 
+ECharts is characterized by its high performance, ease of use, and richness of built-in interactions. 
+Link: [https://echarts.apache.org/en/index.html](https://echarts.apache.org/en/index.html)
+
+#### Chart.js
+Chart.js is an open source diagram library for designers and developers. 
+It is based on JavaScript and supports HTML5 Canvas and JSON. 
+Its high responsiveness makes Chart.js stand out.
+Link: [https://www.chartjs.org](https://www.chartjs.org)
+
+#### D3.js
+D3.js (Data Driven Documents) is a JavaScript-based library that allows you to visualize data with graphical elements in your browser. 
+D3.js is based on the web standards SVG, CSS, and HTML5, which makes D3.js not only fast even with large amounts of data, but also future-proof.
+Link: [https://d3js.org](https://d3js.org)
+
+## Installation
+### How to run the Research Analyzer
+#### Step 1: Install all prerequisites
+1. Install Python.
+2. Install pip.
+3. Install npm or yarn.
+4. [Recommended] Install virtualenv.
+5. [Recommended] If you are using IntelliJ, install the [PyLint Plugin](https://plugins.jetbrains.com/plugin/11084-pylint/).
+
+#### Step 2: Start the backend
+1. Navigate to the backend directory: 
+```
+cd backend
+```
 
-Some good starting resources are:
- - [The Flask mega tutorial](https://blog.miguelgrinberg.com/post/the-flask-mega-tutorial-part-i-hello-world)
- - [Flask Project Website](https://palletsprojects.com/p/flask/)
+2a. [If you have installed virtualenv] Start virtualenv: 
+```
+virtualenv venv
+```
 
-If you don't want to use `virtualenv`, you can install the required dependencies with
+2b. [If you have not installed virtualenv] Install required dependencies: 
```

Comment:
That's actually not correct. You need to do this regardless of the usage of virtualenv.
`virtualenv` just creates a fresh python environment for you so that there are no package conflicts.

But I admit this was also misleading in the previous version of the README.

Reply:
In the `How to modify the Research Analyzer -> Installing new packages` section I already describe how we deal with reinstalling packages. That should do it, right?